### PR TITLE
fix get rook moves

### PR DESCRIPTION
**Pull request by Muhammadbravo**

### CODE REVIEW

File: engine.py [hunk 1]
```diff
@@ -102,8 +117,56 @@ def get_king_moves(self, r, c, moves):
 
 
 	def get_rook_moves(self, r, c, moves):
- 		##TODO
- 		pass
```
Comment (Bilaerl) [inline engine.py:124]: You don't need this line, it is obselete

Comment (Bilaerl) [inline engine.py:124]: You don't need this line, it is obselete

**Review by Bilaerl — CHANGES_REQUESTED** [review]
Nice try
You did not handle the case of the rook jumping over other pieces.
I'll send you Mr. Satatima's solution to bishop moves, since it's kind of similar to the rook's move

File: engine.py (hunk 1 above)
Comment (Muhammadbravo) [reply engine.py:124]: Thank you for that. There's a conflict while i was trying to push to my repo, so I had to merge everything to be able to scale through.

**Comment by Muhammadbravo** [comment]
> Nice try
> You did not handle the case of the rook jumping over other pieces.
> I'll send you Mr. Satatima's solution to bishop moves, since it's kind of similar to the rook's move

Alright, I will await to receive that.

File: engine.py [hunk 2]
```diff
@@ -3,306 +3,319 @@
...
+
+
+
+    def get_rook_moves(self, r, c, moves):
+         direction = (-1, 1) # possible direction
+         
+         if self.light_to_move:
+            
+
+            # if it's light's turn to move
```
Comment (Bilaerl) [inline engine.py]: move comment to line 123 or above it

File: engine.py [hunk 3]
```diff
@@ -3,306 +3,319 @@
...
+                    if (0 <= colnum < len(self.board)): # making sure column is on board
+                        if self.board[r][colnum] == "  ": # if square is empty
+                            moves.append(Move((r,c), (r,colnum), (self.board)))
+                        elif self.board[r][colnum][1] == "d": # if square has opponent piece
+                            moves.append(Move((r,c), (r,colnum), (self.board)))
+                            break
+                        else:
+                            break # when ally piece encountered
+                    else:
+                        break # when off the board
```
Comment (Bilaerl) [inline engine.py:168]: Add a few blank lines under this line to space out the white moves from the dark for readability

**Review by Bilaerl — CHANGES_REQUESTED** [review]
Add a doc-string to the get_rook_moves method

File: engine.py (hunk 3 above)
Comment (Muhammadbravo) [reply engine.py:168]: All solved

**Comment by Muhammadbravo** [comment]
> Add a doc-string to the get_rook_moves method

Alright, will work on that

**Comment by Bilaerl** [comment]
I did not see the changes. In fact your get_rook_moves method is now blank.

**Comment by Bilaerl** [comment]
Work on your commit messages, "update get_rook_moves" is not descriptive enough to give the idea of what the commit is about

**Review by Bilaerl — CHANGES_REQUESTED** [review]
Still
You deleted a whole lot of things.
Maybe you should reset to an earlier commit, then work from there

**Comment by Muhammadbravo** [comment]
> I did not see the changes. In fact your get_rook_moves method is now blank.

I actually don"t know what happened but can you see it now

**Comment by Bilaerl** [comment]
I think you merged my light_team branch into yours

**Comment by Bilaerl** [comment]
There are issues with your code still

**Comment by Bilaerl** [comment]
You save the file without resolving merge conflicts

File: engine.py (hunk 2 above)
Comment (Bilaerl) [reply engine.py]: Move this comment to the same line with if self.light_to_move or to the line above it

**Review by Bilaerl — CHANGES_REQUESTED** [review]
You are almost done, make the changes suggested

**Comment by Bilaerl** [comment]
You merged my branch into yours again
When you make changes you don't have to do anything on the pull request for me to see the changes. Just commit the changes, and push them to your online repo. Github does the rest of the job for you

File: main.py [hunk 4]
```diff
@@ -205,7 +205,11 @@ def animate(move, screen, board, clock):
 		r, c = ((move.start_row + dr*frame/frame_count, move.start_col + dc*frame/frame_count))
 
 		# play sound 
+<<<<<<< HEAD
```
Comment (Bilaerl) [inline main.py]: remove this line

File: main.py [hunk 5]
```diff
@@ -205,7 +205,11 @@ def animate(move, screen, board, clock):
...
+		if not sound_played and ((abs(move.end_row - r) + abs(move.end_col - c))/max((move.end_row+move.end_col+0.01), (r+c+0.01))) < 0.4:
```
Comment (Bilaerl) [inline main.py:208]: remove the '+' on the beginning of this line

File: main.py [hunk 6]
```diff
@@ -205,7 +205,11 @@ def animate(move, screen, board, clock):
...
+=======
 		if not sound_played and ((abs(move.end_row - r) + abs(move.end_col - c))/max(abs(move.end_row+move.end_col + 0.01), abs(r+c + 0.01))) < 0.4:
+>>>>>>> 6557486f6de89b66e5b86c5664758b2f503cd125
```
Comment (Bilaerl) [inline main.py]: remove all of this

**Review by Bilaerl — CHANGES_REQUESTED** [review]
Nice job
You are done with engine.py
I request some changes on main.py which I think are a result of you merging my branch to yours earlier

File: main.py (hunk 4 above)
Comment (Muhammadbravo) [reply main.py]: Alright. On it

File: main.py [hunk 7]
```diff
@@ -205,11 +205,7 @@ def animate(move, screen, board, clock):
 		r, c = ((move.start_row + dr*frame/frame_count, move.start_col + dc*frame/frame_count))
 
 		# play sound 
-<<<<<<< HEAD
-		if not sound_played and ((abs(move.end_row - r) + abs(move.end_col - c))/max((move.end_row+move.end_col+0.01), (r+c+0.01))) < 0.4:
```
Comment (Bilaerl) [inline main.py]: You should have left this one

File: main.py [hunk 8]
```diff
@@ -205,11 +205,7 @@ def animate(move, screen, board, clock):
...
-=======
 		if not sound_played and ((abs(move.end_row - r) + abs(move.end_col - c))/max(abs(move.end_row+move.end_col + 0.01), abs(r+c + 0.01))) < 0.4:
```
Comment (Bilaerl) [inline main.py]: And deleted this one. The first one is the correct one

File: main.py (hunk 7 above)
Comment (Bilaerl) [inline main.py]: You should have left this one

**Review by Bilaerl — CHANGES_REQUESTED** [review]
Even though they look the same, there is a slight difference. And the first one is the correct one

File: main.py (hunk 7 above)
Comment (Bilaerl) [reply main.py]: Just remove the '+' on the beginning of the first one. And delete the second one altogether

**Review by Bilaerl — COMMENTED** [review]
Good Job
Send a pull request to Adebayo

File: main.py (hunk 7 above)
Comment (Muhammadbravo) [reply main.py]: I have removed the second one. The plus sign is showing only here but it is not in the code

**Pull request closed without merging**